api/customers/CustomersApi.php

```diff
@@ -0,0 +1,150 @@
+    }
+
+
+    /**
+     * Searches the customer profiles associated with a Square account using a supported query filter.
+     *
+     * @param       $filter
+     * @param array $sort
+     * @param int   $limit
+     * @return array
+     */
+    public function search($filter, array $sort = ['field' => 'CREATED_AT'], int $limit = 1): array
+    {
+        $search_parameters = [
+            'query' => [
+                'filter' => $filter,
+                'sort'   => $sort,
+            ],
+            'limit' => $limit,
+        ];
+        // Submit the search request.
+        $response = $this->api->sendRequest($search_parameters, $this->post_url . 'search');
+        // Do we have an error ?
+        if (is_array($response) && isset($response['error'])) {
+            return $response;
+        }
+        if (! $response || ! isset($response->customers)) {
+            // Square return an empty body if nothing is found, so we return an empty array.
+            return [];
+        }
+        // Got customer/s, return it.
+        return $response->customers;
+    }
+
+
+    /**
+     * Searches for the customer profile by his email.
+     *
+     * @param string $email
+     * @return array
+     */
+    public function findByEmail(string $email): array
+    {
+        $filter['email_address'] = ['exact' => $email];
+        return $this->search($filter);
+    }
+
+
+    /**
+     * Creates a new customer for a business.
+     *
+     * @return Object|array
+     */
+    public function create()
+    {
+        try {
+            $country_iso = EED_SquareOnsite::getCountryIsoByName($this->billing_info['country']);
+        } catch (\EE_Error | \ReflectionException $e) {
+            $country_iso = '';
+        }
+        $parameters = [
+            'given_name'    => $this->billing_info['first_name'] ?? '',
+            'family_name'   => $this->billing_info['last_name'] ?? '',
+            'email_address' => $this->billing_info['email'] ?? '',
+            'address' => [
+                'address_line_1' => $this->billing_info['address'] ?? '',
+                'address_line_2' => $this->billing_info['address2'] ?? '',
+                'locality'       => $this->billing_info['city'] ?? '',
+                'administrative_district_level_1' => $this->billing_info['state'] ?? '',
+                'postal_code'    => $this->billing_info['zip'] ?? '',
+                'country'        => $country_iso ?? '',
+            ],
+            'phone_number' => $this->billing_info['phone'] ?? '',
+            'reference_id' => (string) $this->primary_registrant->attendee_ID(),
+            'note'         => 'An event attendee'
+        ];
+
+        // Submit the create request.
+        $response = $this->api->sendRequest($parameters, $this->post_url);
+        // Do we have an error ?
+        if (is_array($response) && isset($response['error'])) {
+            return $response;
+        }
+        if (! isset($response->customer)) {
```

Comment:
this is similar to the conditional in `search()`, but that one checks if a response exists at all:

```php
if (! $response || ! isset($response->customers)) {
```

should `! $response ||` be added here as well?

or like suggested above, maybe this should be extracted to its own function?

ex:

```
private function hasCustomerData($response) 
{
	return $response instanceof ???? && isset($response->customers);
}
```

usage:

```suggestion
        if (! $this->hasCustomerData($response->customer)) {
```

other possible names for func:
- hasValidCustomerData
- isValid
- isValidResponse
- ???